cranelift/codegen/src/remove_constant_phis.rs

```diff
@@ -181,11 +231,16 @@ pub fn do_remove_constant_phis(func: &mut Function, domtree: &mut DominatorTree)
     // Phase 1 of 3: for each block, make a summary containing all relevant
     // info.  The solver will iterate over the summaries, rather than having
     // to inspect each instruction in each block.
-    let mut summaries = FxHashMap::<Block, BlockSummary>::default();
+    let bump =
+        Bump::with_capacity(blocks_reverse_postorder.len() * 4 * std::mem::size_of::<Value>());
+    let mut summaries = FxHashMap::with_capacity_and_hasher(
```

Comment:
Does `summaries` still need to be a hashmap? It seems to me like it should be dense. If we can have a default state (either `Option<BlockSummary<'a>>` or a zero-alloc and cloneable empty `BlockSummary`) we could use a `SecondaryMap` here I think.

Reply:
About 30-35% of blocks don't get a block summary (no parameters themselves and no successors have params either) so it isn't a clear win since the hash map miiiiight end up using less space. I'll do a quick test and see if there is anything I can measure.

Reply:
Ah OK, in that case it seems totally fine to stick with the hashmap, unless we're surprised by your results!

Reply:
I didn't get any measurable perf benefits from switching to a `SecondaryMap` but it did actually clean up a bit of `unwrap`s and such, so I do think I'll keep it.